crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,1418 @@
+/// @return A pointer to the new Target
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///
+/// # Safety
+///
+/// The ``description`` type is expected to be a pointer to a null-terminated valid string. If the
+/// pointer is not valid or not null ending, the behavior will be undefined.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_new(description: *mut c_char, num_qubits: usize) -> *mut Target {
+    let description = unsafe {
+        if description.is_null() {
+            None
+        } else {
+            // SAFETY: Per documentation, name points to a null-terminated string of characters.
+            let description: &CStr = CStr::from_ptr(description);
+            Some(description.to_str().unwrap())
+        }
+    };
+
+    let target = Target::new(
+        description.map(|name| name.to_string()),
+        Some(num_qubits),
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )
+    .unwrap();
+    Box::into_raw(Box::new(target))
+}
+
+/// @ingroup QkTarget
+/// Returns the number of qubits of this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return The number of qubits this target can use.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     uint32_t num_qubits = qk_target_num_qubits(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_num_qubits(target: *const Target) -> usize {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    target.num_qubits.unwrap_or_default()
+}
+
+/// @ingroup QkTarget
+/// Returns the description of this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return The description of the Target.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new("New Target", 5);
+///     char *description = qk_target_description(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_description(target: *const Target) -> *mut c_char {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+    if let Some(description) = target.description.as_deref() {
+        CString::new(description).unwrap().into_raw()
+    } else {
+        null_mut()
+    }
+}
+
+/// @ingroup QkTarget
+/// Returns the dt value of this Target.
+///
+/// @param target A pointer to the Target.
+///
+/// @return The dt value of this Target.
+///
+/// # Example
+///     const dt = 10e-9;
+///     QkTarget *target = qk_target_new("New Target", 5, &dt, NULL, NULL, NULL, NULL);
```

Comment:
This example looks out of date.

There are some other cases of this below as well.